### ISSUE TITLE
Permissions on /usr/local/pf/logs aren't correct on Debian

### ISSUE THREAD

**Issue by julsemaan** [inverse-inc/packetfence#3469]
Reported by a user.

He installed PacketFence 8.1.0 on Debian and the rights on /usr/local/pf/logs weren't correct (pf couldn't write), so that caused all log files to fail creation and cause unexpected behavior in the application.

```
$ dpkg -l | grep packetfence
ii packetfence 8.1.0-1 all PacketFence network registration / worm mitigation system
ii packetfence-config 8.1.0-1 all Service use to manage PacketFence configuration.
ii packetfence-doc 8.1.0-1 all documentation for packetfence
ii packetfence-golang-daemon 8.1.0-1 amd64 PacketFence Golang binary.
ii packetfence-ntlm-wrapper 8.1.0-1 amd64 C wrapper around the ntlm_auth utility to log authentication latency and success/failure.
ii packetfence-pfcmd-suid 8.1.0-1 amd64 C wrapper that replace perl-suid dependence
ii packetfence-redis-cache 8.1.0-1 all Init script to manage redis server.
```

**Comment by julsemaan** [comment]
Permissions of the /usr/local/pf directory:
```
ls -la /usr/local/pf

total 8236
drwxr-xr-x 15 pf pf 4096 Aug 15 14:21 .
drwxrwsr-x 12 root staff 4096 Aug 15 14:19 ..
-rw-r--r-- 1 pf pf 1722 Jul 9 20:53 CONTRIBUTING.md
-rw-r--r-- 1 pf pf 18013 Jul 9 20:53 COPYING
-rw-r--r-- 1 pf pf 945 Jul 9 20:53 CREDITS
-rw-r--r-- 1 pf pf 7785247 Jul 9 20:53 ChangeLog
-rw-r--r-- 1 pf pf 115786 Jul 9 20:53 ChangeLog.old
-rw-r--r-- 1 pf pf 4878 Jul 9 20:53 Makefile
-rw-r--r-- 1 pf pf 93307 Jul 9 20:53 NEWS.asciidoc
-rw-r--r-- 1 pf pf 131661 Jul 9 20:53 NEWS.old
-rw-r--r-- 1 pf pf 4041 Jul 9 20:53 README.md
-rw-r--r-- 1 pf pf 281 Jul 9 20:53 README.network-devices
-rw-r--r-- 1 pf pf 73385 Jul 9 20:53 UPGRADE.asciidoc
-rw-r--r-- 1 pf pf 66406 Jul 9 20:53 UPGRADE.old
drwx------ 19 pf pf 4096 Aug 15 14:21 addons
drwxr-xr-x 3 pf pf 4096 Aug 15 14:21 bin
-rw-r--r-- 1 pf pf 0 Jul 9 20:53 build-stamp
drwxrwsr-x 12 pf pf 12288 Aug 15 15:42 conf
-rw-r--r-- 1 pf pf 0 Jul 9 20:53 configure-stamp
drwxr-xr-x 2 pf pf 4096 Aug 15 14:23 db
drwxr-xr-x 12 pf pf 4096 Aug 15 14:21 docs
drwxr-xr-x 26 pf pf 4096 Aug 15 14:21 go
drwxr-xr-x 6 pf pf 4096 Aug 15 14:21 html
drwxr-xr-x 11 pf pf 4096 Aug 15 14:21 lib
dr-xr-xr-x 2 pf pf 4096 Aug 15 15:11 logs
-rw-r--r-- 1 pf pf 468 Jul 9 20:53 packetfence-active-passive.service
-rw-r--r-- 1 pf pf 1614 Jul 9 20:53 packetfence-redis-cache.init
-rw-r--r-- 1 pf pf 809 Jul 9 20:53 packetfence.cron.d
-rw-r--r-- 1 pf pf 2353 Jul 9 20:53 packetfence.init
-rw-r--r-- 1 pf pf 214 Jul 9 20:53 packetfence.journald
-rw-r--r-- 1 pf pf 834 Jul 9 20:53 packetfence.logrotate
-rw-r--r-- 1 pf pf 4502 Jul 9 20:53 packetfence.rsyslog
-rw-r--r-- 1 pf pf 68 Jul 9 20:53 packetfence.rsyslog-drop-in.service
-rw-r--r-- 1 pf pf 334 Jul 9 20:53 packetfence.sudoers
drwxr-xr-x 9 pf pf 4096 Aug 15 14:30 raddb
drwxr-xr-x 2 pf pf 4096 Aug 15 14:21 sbin
drwxr-xr-x 2 pf pf 4096 Aug 15 14:21 src
drwxrwsr-x 17 pf pf 4096 Aug 15 14:58 var
```

**Comment by julsemaan** [comment]
The packaging should have taken care of the rights, but not sure why it failed..

https://github.com/inverse-inc/packetfence/blob/devel/debian/packetfence.postinst#L41

**Comment by danpoltawski** [comment]
I'm the user in question, so let me know if I can be of assistance.

**Comment by MiCRoPhoBIC** [comment]
Looks like I had the same issue:
```
ls -lah /usr/local/pf/|grep logs
dr-xr-xr-x  2 pf   pf     20K Aug 16 13:54 logs
```

**Comment by julsemaan** [comment]
I'm guessing this is a job for the king of the Debian packaging

**Comment by nqb** [comment]
Also see #3145

**Comment by extrafu** [comment]
@fdurand Can you please fix this for v8.2?

**Comment by julsemaan** [comment]
I think this is related to #3749

@nqb could you try again once its part of the builds

**Comment by nqb** [comment]
@julsemaan: I need fingerbank-collector packages for debian devel to test.

**Comment by julsemaan** [comment]
@fdurand, could you look into the fingerbank-collector packaging for Debian.

I'm wild-guessing that Debian is currently untested if the deps aren't satisfied

**Comment by julsemaan** [comment]
We'll likely postpone 8.2 on Debian by a couple of days to have the time to properly package everything and test it right

**Comment by nqb** [comment]
fingerbank-collector 1.1.0 and fingerbank 4.1.2 should now be available for Debian devel.

**Comment by nqb** [comment]
Hello all,

On Debian devel, rights on `logs` directory seems correct:
```
# ls -atld /usr/local/pf/logs/
drwxrwsr-x 2 pf pf 4096 Nov 13 10:02 /usr/local/pf/logs/
```
Strictly identical to CentOS **but** only two files: `packetfence.log` and `httpd.admin.audit.log`. 

After doing this:
```
mkdir /etc/systemd/system/rsyslog.service.d/
cp /etc/systemd/system/syslog.service.d/packetfence.conf /etc/systemd/system/rsyslog.service.d/
systemctl daemon-reload
systemctl status rsyslog
```
I get:
```
# ls -l /usr/local/pf/logs/
total 1268
-rw-r----- 1 root pf   2612 Nov 13 12:53 api-frontend.log
-rw-r----- 1 root pf  18840 Nov 13 13:09 fingerbank.log
-rw-r----- 1 root pf    872 Nov 13 12:48 httpd.aaa.error
-rw-r----- 1 root pf  63889 Nov 13 12:59 httpd.admin.access
-rw-r--r-- 1 pf   pf   1418 Nov 13 12:49 httpd.admin.audit.log
-rw-r----- 1 root pf  34928 Nov 13 12:48 httpd.admin.error
-rw-r----- 1 root pf 818199 Nov 13 12:59 httpd.admin.log
-rw-r----- 1 root pf  27098 Nov 13 13:10 httpd.api-frontend.access
-rw-r----- 1 root pf    414 Nov 13 12:48 httpd.parking.error
-rw-r----- 1 root pf    872 Nov 13 12:49 httpd.portal.catalyst
-rw-r----- 1 root pf    468 Nov 13 12:49 httpd.portal.error
-rw-r----- 1 root pf    912 Nov 13 12:48 httpd.webservices.error
-rw-rw-r-- 1 pf   pf 101530 Nov 13 13:08 packetfence.log
-rw-r----- 1 root pf  45634 Nov 13 12:47 pfconfig.log
-rw-r----- 1 root pf    926 Nov 13 12:48 pfdhcplistener.log
-rw-r----- 1 root pf   1676 Nov 13 12:48 pfdns.log
-rw-r----- 1 root pf  59373 Nov 13 13:10 pfmon.log
-rw-r----- 1 root pf    757 Nov 13 12:48 pfsso.log
-rw-r----- 1 root pf  27791 Nov 13 13:10 pfstats.log
-rw-r----- 1 root pf   3577 Nov 13 12:49 radius-acct.log
-rw-r----- 1 root pf   3577 Nov 13 12:49 radius.log
-rw-r----- 1 root pf   2514 Nov 13 13:06 redis_cache.log
-rw-r----- 1 root pf   3489 Nov 13 13:08 redis_queue.log
-rw-r----- 1 root pf   7833 Nov 13 13:08 redis_server.log

```
versus a CentOS `logs` dir:
```
# ls -l *.log
-rw-rw-r-- 1 root adm    736 Nov 13 06:33 api-frontend.log
-rw-rw-r-- 1 root adm      0 Nov 13 03:09 etcd.log
-rw-rw-r-- 1 root pf   84807 Nov 13 07:11 fingerbank.log
-rw-rw-r-- 1 root adm      0 Nov 13 03:09 haproxy_portal.log
-rw-rw-r-- 1 pf   pf       0 Nov 13 03:09 httpd.admin.audit.log
-rw-rw-r-- 1 pf   pf       0 Nov 13 03:09 httpd.admin.log
-rw-rw-r-- 1 root pf       0 Nov 13 03:09 mariadb_error.log
-rw-rw-r-- 1 pf   pf    6533 Nov 13 07:06 packetfence.log
-rw-rw-r-- 1 pf   pf       0 Nov 13 03:09 pfconfig.log
-rw-rw-r-- 1 pf   pf    3358 Nov 13 07:03 pfdhcplistener.log
-rw-rw-r-- 1 pf   pf       0 Nov 13 03:09 pfdns.log
-rw-rw-r-- 1 root pf       0 Nov 13 03:09 pffilter.log
-rw-rw-r-- 1 pf   pf  660191 Nov 13 07:11 pfmon.log
-rw-rw-r-- 1 root adm      0 Nov 13 03:09 pfsso.log
-rw-rw-r-- 1 root adm 262919 Nov 13 07:11 pfstats.log
-rw-rw-r-- 1 root pf       0 Nov 13 03:09 radius-acct.log
-rw-rw-r-- 1 root pf       0 Nov 13 03:09 radius.log
-rw-rw-r-- 1 pf   pf   18396 Nov 13 07:06 redis_cache.log
-rw-rw-r-- 1 root pf   18918 Nov 13 07:10 redis_queue.log
-rw-rw-r-- 1 root adm  48954 Nov 13 07:10 redis_server.log
-rw-rw-r-- 1 pf   pf       0 Nov 13 03:09 violation.log
```

I will push a fix for rsyslog.

**Comment by nqb** [comment]
@fdurand and @julsemaan: I'm a bit surprised by permissions on logs files for Debian regarding permissions on CentOS. But that seems to work because all files has non-zero size.

What's your point of view ?

**Comment by julsemaan** [comment]
Actually, the rights should be fine on both distros now if you're running the latest devel

Its not normal that they are wrong on CentOS for you give this has been fixed in 8.2

**Comment by nqb** [comment]
As discussed with @julsemaan, rights on files have been fixed for all distros by 71c83c6.

I close the issue.
